boggle/Boggle_works.py

```diff
@@ -0,0 +1,152 @@
+		self.setup()
+
+##the boggle gui setup, still need to fix the 'shake' cmd so that it resets the board		
+    def setup(self):
+		global a
+		
+		shuffle(a)
+		self.ca_width = 40
+		self.ca_height = 40
+		self.gr(4, side=LEFT, bd=10)
+        # left frame
+		for i in range(0,16):
+			self.fr(side=LEFT, bd=5)
+			self.canvas = self.ca(width=self.ca_width, height=self.ca_height, bg='white')
+			item1 = self.canvas.text([0,0], self.shake(i), font = 24) 
+			self.endfr()
+
+		self.endgr()
+
+		self.fr(RIGHT, bd=5)
+		self.bu(text='Shake', command=self.setup)
+		self.bu(text='Words', command=proc_words)
+		self.endfr()
+		
+
+    def shake(self,b):
+        global a
+        global letters
+
+
+##the letter blocks of a standard boggle game
+        blocks = {1:['M','H','QU','U','I','N'],2:['E','E','N','G','A','A'],
+              3:['A','S','F','P','F','K'],4:['E','N','U','I','E','S'],
+              5:['O','B','J','O','A','B'],6:['L','T','Y','E','R','T'],
+              7:['A','S','H','P','O','C'],8:['G','E','W','H','E','N'],
+              9:['Y','T','D','T','S','I'],10:['A','O','T','W','T','O'],
+              11:['L','N','H','N','Z','R'],12:['I','L','R','E','X','D'],
+              13:['V','Y','D','R','E','L'],14:['W','R','T','E','H','V'],
+              15:['S','S','I','O','T','E'],16:['O','U','M','C','I','T']}
+
+
+        block = a[b]
+        side = blocks[block]
+        letter = choice(side)
+        letters.append(letter)
+        if len(letters) > 16:
+            del letters[0]
+        return letter
+
+				
+def proc_words():
+	global letters
+	print letters
+	fil = 'C:\Users\provo\SD\Boggle\words.txt'
+	try:
+		 doc = open(fil, 'r')
+	except KeyError:
+		return "file not found"
+
+##opens the dictionary file, processes words, eliminates 2 letter words and words > 16 letters
+		
+	for word in doc:
+		word = string.rstrip(word)
+		word = string.upper(word)
+		
+		if len(word) < 3 or len(word)>16:
+			continue
+
+##this takes each word and checks to see if it's letters match those in the grid, special handling for the 'qu' block  			
+		else:
+			lets = list(letters)
+			q_test = 0
+			wrd = []
+			qt = False
+			for letter in word:
+				if letter =='Q':
+					letter = 'QU'
+					q_test = 1
+					qt = True
+				elif letter =='U' and qt:
+					qt = False
+					continue
+				if letter in lets:
+					lets.remove(letter)
+					wrd.append(letter)
+				else:
+					break
+
+#if all the words letters are in the grid, send it the pathing function
+
+			if len(wrd) + q_test == len(word):
+				nodes = extra(letters,wrd[0])
+				for node in nodes:
+					if find_words(wrd, node,[node]):
+						winners.append(word)
+						break
+			
+	print winners, len(winners)
+
+
+##pathing function that takes the word as a list of letters, the node of the first letter and the path traversed			
+def find_words(word, node, path):
+	global letters
+	global connect
+	touches = connect[node]
+	dist = len(path)
+	stop = len(word)
+	if dist == stop:

```

Comment:
Add a comment about this base case

   # Successfully reached the end of the word.